src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.macOS.cs

```diff
@@ -1,17 +1,164 @@
-﻿using Windows.UI.Xaml.Controls;
+﻿using Uno.Extensions;
+using Uno.UI.DataBinding;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
 using System;
 using System.Collections.Generic;
-using System.Linq;
+using System.Drawing;
+using Uno.Disposables;
+using System.Runtime.CompilerServices;
 using System.Text;
-using System.Threading.Tasks;
-using Windows.UI.Xaml.Data;
-using Windows.UI.Xaml.Media;
-using Windows.UI.Xaml.Media.Animation;
+using System.Linq;
+
+using Foundation;
+using CoreGraphics;
+using Uno.UI.Extensions;
 
 namespace Windows.UI.Xaml.Controls
 {
-	public partial class ScrollViewer 
+	public partial class ScrollViewer : ContentControl
 	{
-		private void UpdateZoomedContentAlignment() { }
+		/// <summary>
+		/// On iOS 10-, we set a flag on the view controller such that the CommandBar doesn't automatically affect ScrollViewer content 
+		/// placement. On iOS 11+, we set this behavior on the UIScrollView itself.
+		/// </summary>
+		internal static bool UseContentInsetAdjustmentBehavior => false;
+
+		/// <summary>
+		/// The <see cref="UIScrollView"/> which will actually scroll. Mostly this will be identical to <see cref="_sv"/>, but if we're inside a
+		/// multi-line TextBox we set it to <see cref="MultilineTextBoxView"/>.
+		/// </summary>
+		private IUIScrollView _scrollableContainer;
+
+		partial void OnApplyTemplatePartial()
+		{
+			SetScrollableContainer();
+		}
+
+		protected override void OnLoaded()
+		{
+			SetScrollableContainer();
+			base.OnLoaded();
+		}
+
+		private void SetScrollableContainer()
+		{
+			_scrollableContainer = _sv;
+
+#if false // TODO Scrollable textbox https://github.com/unoplatform/uno/issues/626
+			if (this.FindFirstParent<TextBox>() != null)
+			{
+				var multiline = this.FindFirstChild<TextBoxView>();
+				if (multiline != null)
+				{
+					_scrollableContainer = multiline;
+				}
+			}
+#endif
+		}
+
+		partial void ChangeViewScroll(double? horizontalOffset, double? verticalOffset, bool disableAnimation)
+		{
+			if (_scrollableContainer != null)
+			{
+				// iOS doesn't limit the offset to the scrollable bounds by itself
+				var point = new CGPoint(horizontalOffset ?? HorizontalOffset, verticalOffset ?? VerticalOffset);
+
+				var newOffset = point.Clamp(CGPoint.Empty, _scrollableContainer.UpperScrollLimit);
+
+				_scrollableContainer.SetContentOffset(newOffset, !disableAnimation);
+			}
+		}
+
+		partial void OnZoomModeChangedPartial(ZoomMode zoomMode)
+		{
+			// On iOS, zooming is disabled by setting Minimum/MaximumZoomScale both to 1
+			switch (zoomMode)
+			{
+				case ZoomMode.Disabled:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty 'case' clause.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5303491)